foursight_core/checks/access_key_expiration_detection.py

```diff
@@ -78,19 +93,24 @@ def refresh_access_keys(connection, **kwargs):
         access_key_res = post_metadata(access_key_req, 'access-keys', key=connection.ff_keys)
         access_key_id = access_key_res.get('access_key_id')
         secret_access_key = access_key_res.get('secret_access_key')
-        import pdb ; pdb.set_trace()
         if not access_key_id or not secret_access_key:
             # We will say these must occur in pairs; both at the top level or both within the @graph array.
             graph_item = access_key_res.get('@graph', [{}])[0]
             access_key_id = graph_item.get('access_key_id')
             secret_access_key = graph_item.get('secret_access_key')
         s3_obj = {'secret': secret_access_key, 'key': access_key_id, 'server': s3.url}
+        # Now we store this newly generated access key in a (secure bucket) in S3,
+        # e.g. s3://elasticbeanstalk-fourfront-mastertest-system/access_key_foursight
         s3.s3_put_secret(json.dumps(s3_obj), kp_name)
         full_output['successfully_generated'].append(email)
-        # clear out old keys after generating new one
+        # Delete any old access keys after generating a new one (see VERY IMPORTANT comment above).
```

Comment:
if i understand what you're saying in all these comments, and it seems pretty clear: should we then affirmatively double-check that we can access under a different key and that the key being deleted is in fact a different uuid, etc?

Reply:
Yes